en/development/testing.rst

```diff
@@ -1322,6 +1322,281 @@ comparison files as they are referenced:
     #
     #   modified:   tests/comparisons/example.php
 
+Console Integration Testing
+===========================
+
+To make testing console applications easier, CakePHP comes with a
+``ConsoleIntegrationTestCase`` class that can be used to test console applications
+and assert against their results.
+
+.. versionadded:: 3.5.0
+
+    The ``ConsoleIntegrationTestCase`` was added.
+
+To get started testing your console application, create a test case that extends
+``Cake\TestSuite\ConsoleIntegrationTestCase``. This class contains a method
+``exec()`` that is used to execute your command. You can pass the same string
+you would use in the CLI to this method.
+
+Let's start with a very simple shell, located in **src/Shell/MyConsoleShell.php**::
+
+    namespace App\Shell;
+
+    use Cake\Console\ConsoleOptionParser;
+    use Cake\Console\Shell;
+
+    class MyConsoleShell extends Shell
+    {
+        public function getOptionParser()
+        {
+            $parser = new ConsoleOptionParser();
+            $parser->setDescription('My cool console app');
+            return $parser;
+        }
+    }
+
+To write an integration test for this shell, we would create a test case in
+**tests/TestCase/Shell/MyConsoleShellTest.php** that extends
+``Cake\TestSuite\ConsoleIntegrationTestCase``. This shell doesn't do much at the
+moment, but let's just test that our shell's description is displayed in ``stdout``::
+
+    namespace App\Test\TestCase\Shell;
+
+    use Cake\TestSuite\ConsoleIntegrationTestCase;
+
+    class MyConsoleShellTest extends ConsoleIntegrationTestCase
+    {
+        public function testDescriptionOutput()
+        {
+            $this->exec('my_console');
```

Comment:
It does not, apparently. I created these shells and test case when writing this to ensure they worked like I'm promising here.

The shell seems to skip startup/main since it doesn't exist: https://api.cakephp.org/3.4/source-class-Cake.Console.Shell.html#468-475

And just outputs `$this->OptionParser->help($command)` which in turn outputs `HelpFormatter::text` which includes the description.